base/binaryplatforms.jl

```diff
@@ -171,6 +178,7 @@ end
 
 # Allow us to easily serialize Platform objects
 function Base.show(io::IO, p::Platform)
+    os(p) == "any" && return print(io, "any")
```

Comment:
I don't think AnyPlatform needs special treatment? I don't think Pkg ever deals directly with platforms defined elsewhere. AnyPlatform is only used internally in BinaryBuilder.

Reply:
We are not talking about Pkg here. This about Base and Artifacts as a standard library built into the system image.

Pkg, like BinaryBuilderBase, introduces new `AbstractPlatform` subtypes causing invalidations due to the use of `Dict{AbstractPlatform}` in Artifacts.jl.

As you pointed out, BinaryBuilderBase.jl does call `artifacts_meta` in Artifacts, possibly passing in `AnyPlatform()` which this PR would convert to `Platform("any", "any")`.

https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/b1d6c16dca0b0c6adab208ab1f7056777be7defc/src/Prefix.jl#L751

Showing `Platform("any", "any")` does throw an error though which is what I'm trying to handle here.

Reply:
> As you pointed out, BinaryBuilderBase.jl does call `artifacts_meta` in Artifacts, possibly passing in `AnyPlatform()` which this PR would convert to Platform("any", "any").

No, we pass a concrete target platform, which is a `Platform` object, never `AnyPlatform`.  I still don't think `AnyPlatform` needs special treatment.  My only point in https://github.com/JuliaLang/julia/pull/54073#issuecomment-2056928089 was not to remove `AbstractPlatform`.